scipy/optimize/_qap.py

```diff
@@ -0,0 +1,420 @@
+import numpy as np
+import operator
+from . import linear_sum_assignment, minimize_scalar, OptimizeResult
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    maximize=False,
+    options=None
+):
+    r"""
+    Solve the quadratic assignment problem.
+
+    This function solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem (GMP) using the Fast Approximate QAP Algorithm
+    (FAQ) [1]_.
+
+    Quadratic assignment solves problems of the following form:
+
+    .. math::
+
+        \min_P & \ {\ \text{trace}(A^T P B P^T)}\\
+        \mbox{s.t. } & {P \ \epsilon \ \mathcal{P}}\\
+
+    where :math:`\mathcal{P}` is the set of all permutation matrices,
+    and :math:`A` and :math:`B` are adjacency matrices.
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, non-negative
+        The square adjacency matrix :math:`A` in the objective function above.
+
+    dist_matrix : 2d-array, square, non-negative
+        The square adjacency matrix :math:`B` in the objective function above.
+
+    partial_match : 2d-array of integers, optional, (default = None)
+        Allows the user to fix part of the matching between the two adjacency
+        matrices. In the literature, a partial match is also known as a
+        "seed" [2]_.
+
+        Each row of `partial_match` specifies the indices of a pair of
+        corresponding nodes, that is, vertex ``partial_match[i, 0]`` is
+        matched to vertex ``partial_match[i, 1]``. Accordingly,
```

Comment:
Oops this didn't make this into asaadeldin11/scipy#1
```suggestion
        corresponding nodes, that is, node``partial_match[i, 0]`` of
        `cost_matrix`  is matched to node ``partial_match[i, 1]`` of
        `dist_matrix`. Accordingly,
```